File: GFG/Find patterns/Find patterns.java
class Solution{
    static int numberOfSubsequences(String S, String W){
        // code here
        int ans=0;
        boolean[]vis=new boolean[S.length()];
        for(int i=0;i<S.length();i++){
            if(S.charAt(i)==W.charAt(0) && !vis[i]){
                vis[i]=true;
                int j=i+1;
                int k=1;
                while(j<S.length() && k<W.length()){
                    if(S.charAt(j)==W.charAt(k) && !vis[j]){
                        vis[j]=true;
                        k++;
                    }
                    j++;
                }
                
                if(k==W.length()){
                    ans++;
                }else{
                    break;
                }
            }
        }
        return ans;
    }
}

File: GFG/Find patterns/Find patterns.cpp
class Solution{   
public:
    int numberOfSubsequences(string S, string W){
        // code here 
        int k,c=0;
        for(int i=0;i<S.size();i++) {
            if(S[i]==W[0]) {
                k=0;
                for(int j=0;j<S.size() && k<W.size();j++) {
                    if(S[j]==W[k])
                    { S[j]='@'; k++; }
                }
                if(k==W.size()) c++;
            }
        }
        return c;
    }
};
